overlord/devicestate/devicestate.go

```diff
@@ -719,10 +724,9 @@ func remodelTasks(ctx context.Context, st *state.State, current, new *asserts.Mo
 			if !isNotInstalled(err) {
 				return nil, err
 			}
-			if modelSnap.Presence == "required" {
-				needsInstall = true
-			}
+			needsInstall = true
```

Comment:
maybe the fix would be here?
```
if modelSnap.Presence == "required" {
  needsInstall = true
} else {
   continue
}
```

not sure though as the loop is complex

Reply:
You're right that we want to consider a channel change. I'll fix that.

Reply:
@pedronis that fix looks to do what we want, thanks for the suggestion. I also agree that this loop is complex, and I think there is room for improvement. Would you suggest doing that in this PR, or in a separate one?

Reply:
I pushed that loop refactor, can always revert if we don't want it now.